src/compiler/checker.ts

```diff
@@ -22057,7 +22057,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             let generalizedSource = source;
             let generalizedSourceType = sourceType;
 
-            if (isLiteralType(source) && !typeCouldHaveTopLevelSingletonTypes(target)) {
+            if (isLiteralType(source) && (!typeCouldHaveTopLevelSingletonTypes(target) && !(target.flags & TypeFlags.Never))) {

```

Comment:
```suggestion
            if (!(target.flags & TypeFlags.Never) && isLiteralType(source) && !typeCouldHaveTopLevelSingletonTypes(target)) {

```

Reply:
I've added your suggestion manually, because I also cann't commit the suggestion using github ui.